app-libs/stf/src/trusted_call.rs

```diff
@@ -18,39 +18,43 @@
 #[cfg(feature = "evm")]
 use sp_core::{H160, H256, U256};
 
-use crate::{
-	best_energy_helpers::storage::merkle_roots_map_key, helpers::ensure_enclave_signer_account,
-	StfError, TrustedOperation,
-};
-use binary_merkle_tree::merkle_root;
-use codec::{alloc::sync::Arc, Decode, Encode};
+#[cfg(feature = "evm")]
+use std::vec::Vec;
+
+use crate::{helpers::ensure_enclave_signer_account, StfError, TrustedOperation};
+use codec::{Compact, Decode, Encode};
 use frame_support::{ensure, traits::UnfilteredDispatchable};
+#[cfg(feature = "evm")]
+use ita_sgx_runtime::{AddressMapping, HashedAddressMapping};
 pub use ita_sgx_runtime::{Balance, Index};
 use ita_sgx_runtime::{Runtime, System};
 use itp_node_api::metadata::{provider::AccessNodeMetadata, NodeMetadataTrait};
-use itp_node_api_metadata::pallet_enclave_bridge::EnclaveBridgeCallIndexes;
-use itp_stf_interface::ExecuteCall;
-use itp_stf_primitives::types::{AccountId, KeyPair, OrdersString, ShardIdentifier, Signature};
-use itp_types::OpaqueCall;
+use itp_node_api_metadata::{
+	pallet_balances::BalancesCallIndexes, pallet_enclave_bridge::EnclaveBridgeCallIndexes,
+	pallet_proxy::ProxyCallIndexes,
+};
+use itp_stf_interface::{ExecuteCall, SHARD_VAULT_KEY};
+use itp_stf_primitives::types::{AccountId, KeyPair, ShardIdentifier, Signature};
+use itp_types::{parentchain::ProxyType, Address, OpaqueCall};
 use itp_utils::stringify::account_id_to_string;
 use log::*;
-use simplyr_lib::{pay_as_bid_matching, MarketInput, MarketOutput, Order};
 use sp_io::hashing::blake2_256;
-use sp_runtime::{
-	traits::{Keccak256, Verify},
-	MultiAddress,
-};
-#[cfg(feature = "evm")]
-use std::vec::Vec;
-use std::{format, fs, prelude::v1::*, time::Instant};
-
-#[cfg(feature = "evm")]
-use ita_sgx_runtime::{AddressMapping, HashedAddressMapping};
+use sp_runtime::{traits::Verify, MultiAddress};
+use std::{format, prelude::v1::*, sync::Arc};
 
 #[cfg(feature = "evm")]
 use crate::evm_helpers::{create_code_hash, evm_create2_address, evm_create_address};
+use crate::helpers::get_storage_by_key_hash;
 
-use crate::best_energy_helpers::{write_orders, write_results, ORDERS_DIR, RESULTS_DIR};
+// Group imports that are for OLI to make upstream merges easier.
+use crate::best_energy_helpers::{
+	storage::merkle_roots_map_key, write_orders, write_results, ORDERS_DIR, RESULTS_DIR,
+};
+use binary_merkle_tree::merkle_root;
+use itp_stf_primitives::types::OrdersString;
+use simplyr_lib::{pay_as_bid_matching, MarketInput, MarketOutput, Order};
+use sp_runtime::traits::Keccak256;
+use std::{fs, time::Instant};
```

Comment:
Re-organised the import here that are used in the OLI case only, such that upstream merges are a bit easier.